transport/client.go

```diff
@@ -131,5 +241,18 @@ func (c *Client) Connect() error {
 			}
 		}
 	}()
-	return nil
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
This Ticker/For loop seems sub-optimal. It'll set the minimum time it takes to connect to at least a second. I would create a channel at the start of this function, and then close that channel after calling `c.state.SetComplete`. We can then wait on that channel down here rather than using a ticker.